mountpoint-s3/src/prefetch/part_queue.rs

```diff
@@ -42,7 +44,7 @@ impl<E: std::error::Error + Send + Sync> PartQueue<E> {
     /// empty.
     ///
     /// If this method returns an Err, the PartQueue must never be accessed again.
-    pub async fn read(&self, length: usize) -> Result<Part, PrefetchReadError<E>> {
+    pub async fn read(&self, length: usize) -> ObjectClientResult<Part, PrefetchReadError, E> {
```

Comment:
Still think this error type is inverted; it expands to
```
Result<Part, ObjectClientError<PrefetchReadError, E>>
```
which seems wrong: `ObjectClientError` is about errors in the `ObjectClient`, but the prefetcher lives higher in the stack than that.

Reply:
You're right. Inverted now.